### PR TITLE
YJDH-676: Fetch and autofill employee data

### PR DESCRIPTION
**Pull request by melniiv**
## Description :sparkles:
Fetch employee data by Employee name and voucher serial number

image

### CODE REVIEW

**Comment by terovirtanen** [comment]
**API branch is deployed to platta: https://kesaseteli-pr2093.api.dev.hel.ninja/healthz** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**HANDLER branch is deployed to platta: https://handler-pr2093.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://handler-pr2093.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
**YOUTH branch is deployed to platta: https://youth-pr2093.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**EMPLOYER branch is deployed to platta: https://kesaseteli-pr2093.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://kesaseteli-pr2093.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://youth-pr2093.dev.hel.ninja :laughing::tada::tada::tada:

File: frontend/kesaseteli/employer/src/components/application/steps/step2/accordions/EmploymentAccordion.tsx [hunk 1]
```diff
@@ -29,6 +33,27 @@ type Props = {
...
+    enableFetchEmployeeDataButton();
+  }, []);
+
+  const enableFetchEmployeeDataButton = () => {
+    const formDataVoucher = getValues().summer_vouchers[index];
+    setIsFetchEmployeeDataEnabled(
+      formDataVoucher.employee_name.length > 0 &&
+        formDataVoucher.summer_voucher_serial_number.length > 0
+    );
+  };
```
Comment (nikomakela) [inline frontend/kesaseteli/employer/src/components/application/steps/step2/accordions/EmploymentAccordion.tsx:51]: They should be other way around, so I would know what is called.

File: frontend/kesaseteli/employer/src/components/application/steps/step2/accordions/EmploymentAccordion.tsx [hunk 2]
```diff
@@ -29,6 +33,27 @@ type Props = {
...
+
+  useEffect(() => {
+    enableFetchEmployeeDataButton();
+  }, []);
+
+  const enableFetchEmployeeDataButton = () => {
+    const formDataVoucher = getValues().summer_vouchers[index];
+    setIsFetchEmployeeDataEnabled(
+      formDataVoucher.employee_name.length > 0 &&
+        formDataVoucher.summer_voucher_serial_number.length > 0
```
Comment (nikomakela) [inline frontend/kesaseteli/employer/src/components/application/steps/step2/accordions/EmploymentAccordion.tsx:49]: Checking with mobile, so asking, can it be null?

```suggestion
formDataVoucher?.employee_name?.length && formDataVoucher?.summer_voucher_serial_number?.length
```

**Review by nikomakela — COMMENTED** [review]
Looking good. Left a couple of trivial comments. Should the API be deployed first / together with this? I could recheck this with computer later.

File: frontend/kesaseteli/employer/src/components/application/steps/step2/accordions/EmploymentAccordion.tsx (hunk 2 above)
Comment (melniiv) [reply frontend/kesaseteli/employer/src/components/application/steps/step2/accordions/EmploymentAccordion.tsx:49]: No it cannot be. Button is shown only as part of the form, and form is shown only when employment voucher is added to the application in backend and returned, so we have already form data. Name is always empty string in the beginning, the same as serial number.

File: frontend/kesaseteli/employer/src/components/application/steps/step2/accordions/EmploymentAccordion.tsx (hunk 1 above)
Comment (melniiv) [reply frontend/kesaseteli/employer/src/components/application/steps/step2/accordions/EmploymentAccordion.tsx:51]: i dont understand what you mean

**Comment by melniiv** [comment]
Fixed linting issues, but youth still might fail test, not sure what is that, not related to my task. I think Kari merged at least to develop branch all latest backend changes. Yes, should be installed with frontend to production.

**Comment by terovirtanen** [comment]
**HANDLER branch is deployed to platta: https://handler-pr2093.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**EMPLOYER branch is deployed to platta: https://kesaseteli-pr2093.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://handler-pr2093.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://kesaseteli-pr2093.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://youth-pr2093.dev.hel.ninja :laughing::tada::tada::tada:

File: frontend/kesaseteli/employer/src/components/application/steps/step2/accordions/EmploymentAccordion.tsx (hunk 1 above)
Comment (nikomakela) [reply frontend/kesaseteli/employer/src/components/application/steps/step2/accordions/EmploymentAccordion.tsx]: yep, that I meant

**Comment by terovirtanen** [comment]
**API branch is deployed to platta: https://kesaseteli-pr2093.api.dev.hel.ninja/healthz** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**HANDLER branch is deployed to platta: https://handler-pr2093.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://kesaseteli-pr2093.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://youth-pr2093.dev.hel.ninja :laughing::tada::tada::tada:

**Review by nikomakela — APPROVED** [review]

**Pull request merged**